cranelift/codegen/src/licm.rs

```diff
@@ -87,7 +88,33 @@ fn create_pre_header(
     {
         // We only follow normal edges (not the back edges)
         if !domtree.dominates(header, last_inst, &func.layout) {
-            func.change_branch_destination(last_inst, pre_header);
+            match func.dfg.analyze_branch(last_inst) {
+                // Change branch destination to preheader.
+                BranchInfo::SingleDest(..) => func.change_branch_destination(last_inst, pre_header),
```

Comment:
It seems we could fold this into a helper alongside `change_branch_destination` on `Function`, by defining a `rewrite_branch_destination(inst, old_target, new_target)` that replaces any `old_target` target in the instruction with `new_target`, doing the replacement on any target(s) that exist in the instruction. Thoughts? This would also remove the awkward match below.

Reply:
Exactly! Thanks for pointing out.
I'll add `Function::rewrite_branch_destination`.